src/SplitCore.hs

```diff
@@ -1,8 +1,22 @@
+{-|
+Module           : SplitCore
+Description      : An intermediate datatype for partially expanded expressions.
+
+'SplitCore' is an intermediary between 'Syntax' (what the user types) and 'Core'
```

Comment:
I wouldn't characterize it as an intermediary - it's got basically nothing to do with `Syntax`. I'd instead characterize it as a partially-constructed AST in which zero or more sub-trees may be absent that's represented as an explicit pointer graph in the `Map` component.